src/views/Player.vue

```diff
@@ -105,15 +115,32 @@ export default defineComponent({
       hasQuizEnded: false, // whether the quiz has ended - only valid for quizType = assessment
       sessionId: "", // id of the session created for a user-quiz combination
     });
+
+    const requiresAuth = computed(() => {
+      return props.userId != null && props.apiKey != null;
+    });
+
```

Comment:
this is already done in `router/index.ts` no? There is no reason to repeat it here.